File: openapi/examples/typescript/health.ts
// GET /v1/health - service liveness, uptime, API/rules versions, and
// non-sensitive source freshness. No authentication required, not billed.
//
// Run with: npx tsx health.ts
export interface SourceFreshness {
  key: string;
  name?: string | null;
  data_status: "current" | "stale" | "needs_review" | "unavailable";
  last_refreshed_at?: string | null;
}

export interface HealthResponse {
  status: "ok" | "degraded";
  uptime_seconds: number;
  api_version: string;
  rules_version?: string | null;
  sources?: SourceFreshness[];
}

async function main() {
  // No auth headers required for /v1/health, on either host.
  const res = await fetch("https://api.aduatlas.example.com/v1/health");
  const health = (await res.json()) as HealthResponse;

  console.log(`status: ${health.status} (api ${health.api_version})`);
  for (const source of health.sources ?? []) {
    console.log(`  ${source.key}: ${source.data_status}`);
  }
}

main();

File: openapi/examples/python/health.py
"""GET /v1/health - service liveness, uptime, API/rules versions, and
non-sensitive source freshness. No authentication required, not billed.

Run: python health.py
Requires: pip install httpx
"""
from __future__ import annotations

import httpx


def main() -> None:
    # No auth headers required for /health, on either host.
    with httpx.Client(timeout=15.0) as client:
        resp = client.get("https://api.aduatlas.example.com/v1/health")
    resp.raise_for_status()
    health = resp.json()

    print(f"status: {health['status']} (api {health['api_version']})")
    for source in health.get("sources", []):
        print(f"  {source['key']}: {source['data_status']}")

    # Also reachable through the RapidAPI gateway (the Hub-registered path
    # has no /v1 prefix); still no auth headers required.
    with httpx.Client(timeout=15.0) as client:
        resp = client.get("https://property-feasibility4.p.rapidapi.com/health")
    resp.raise_for_status()


if __name__ == "__main__":
    main()
